app/js/lib/foxbox/services/base.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const TYPE = 'Unknown';
+
+const p = Object.freeze({
+  // Private properties.
+  net: Symbol('net'),
+  settings: Symbol('settings'),
+
+  // Public getter only properties.
+  id: Symbol('id'),
+  getters: Symbol('getters'),
+  setters: Symbol('setters'),
+
+  // Private methods.
+  getChannel: Symbol('getChannel'),
+  getSetterValueType: Symbol('getSetterValueType'),
+});
+
+export default class BaseService {
+  constructor(props, config) {
+    // Private properties.
+    this[p.net] = config.net;
+    this[p.settings] = config.settings;
+
+    // Public getter only properties.
+    this[p.id] = props.id;
+    this[p.getters] = props.getters;
+    this[p.setters] = props.setters;
+
+    // Public properties.
+    this._tags = props.tags;
+  }
+
+  // Public getter only properties.
+  get type() {
+    return TYPE;
+  }
+
+  get id() {
+    return this[p.id];
+  }
+
+  get name() {
+    return '';
+  }
+
+  get hasGetters() {
+    return Object.keys(this[p.getters]).length > 0;
+  }
+
+  get hasSetters() {
+    return Object.keys(this[p.setters]).length > 0;
+  }
+
+  // Public methods.
+  /**
+   * Call a service setter with a value.
+   *
+   * @param {Object} selector
+   * @param {string|number|boolean} value
+   * @return {Promise}
+   */
+  set(selector, value = '') {
+    const setter = this[p.getChannel](this[p.setters], selector);
+    const setterType = this[p.getSetterValueType](setter.kind);
+
+    return this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/channels/set`,
+      'PUT',
+      // Query operation by id.
+      [[{ id: setter.id }, { [setterType]: value }]]
+    );
+  }
+
+  /**
+   * Call a service getter.
+   *
+   * @param {Object} selector
+   * @return {Promise}
+   */
+  get(selector) {
+    const getter = this[p.getChannel](this[p.getters], selector);
+    const body = { id: getter.id };
+
+    if (getter.kind.type === 'Binary') {
+      return this[p.net].fetchBlob(
+        `${this[p.net].origin}/api/v${this[p.settings].apiVersion}` +
+        '/channels/get',
+        // For now we only support JPEG blobs.
```

Comment:
__issue__: hmm, looks inappropriate for the base class getter... Does your lights getter work correctly with this? Maybe we should have optional ```mimeType``` for the ```get``` method?